src/lookup-table/models/LookupTable.model.ts

```diff
@@ -47,69 +63,107 @@ const LookupTableModel = () => {
         `,
         { name: lookupTableName }
       )
-      return data.lookupTables.totalCount
+
+      return result.lookupTables.totalCount
     } catch (error) {
       throw error
     }
   }
 
-  const createTable = async ({ tableName, fieldMap: fieldMaps }: LookupTableStructurePropType) => {
-    const text = `CREATE TABLE lookup_table_${tableName}
+  const createTable = async ({
+    name: tableName,
+    fieldMap: fieldMaps,
+  }: LookupTableStructureType): Promise<boolean> => {
+    try {
+      const text = `CREATE TABLE lookup_table_${tableName}
       (
         ${fieldMaps.map((fieldMap) => `${fieldMap.fieldname} ${fieldMap.dataType}`).join(', ')}
       )`
 
-    const result = await DBConnect.query({ text })
-    return result.rows
+      await DBConnect.query({ text })
+      return true
+    } catch (error) {
+      throw error
+    }
   }
 
-  const createRow = async ({ tableName, row }: { tableName: string; row: any }) => {
-    const text = `INSERT INTO lookup_table_${tableName}(${Object.keys(row)}) VALUES (
+  const createRow = async ({
+    tableName,
+    row,
+  }: {
+    tableName: string
+    row: any
+  }): Promise<{ id: string }[]> => {
+    try {
+      const text = `INSERT INTO lookup_table_${tableName}(${Object.keys(row)}) VALUES (
           ${Object.keys(row)
             .map((key, index) => {
               return '$' + String(index + 1)
             })
             .join(', ')}) RETURNING id`
-    const result = await DBConnect.query({ text, values: [...Object.values(row)] })
-    return result.rows.map((row: any) => row.id)
-  }
 
-  const updateRow = async ({ tableName, row }: { tableName: string; row: any }) => {
-    let primaryKeyIndex = 0
-    const setText = Object.keys(row)
-      .map((key, index) => {
-        const currentIndex = index + 1
-        if (key === 'id') {
-          primaryKeyIndex = currentIndex
-          return ''
-        }
-
-        return `${key} = $${currentIndex}`
+      const result: QueryResult<{ id: number }> = await DBConnect.query({
+        text,
+        values: [...Object.values(row)],
       })
-      .filter(Boolean)
-      .join(', ')
 
-    const text = `UPDATE lookup_table_${tableName} SET
-    ${setText} WHERE id = $${primaryKeyIndex}`
+      return result.rows.map((row: any) => row.id)
+    } catch (error) {
+      throw error
+    }
+  }
+
+  const updateRow = async ({
+    tableName,
+    row,
+  }: {
+    tableName: string
+    row: any
+  }): Promise<boolean> => {
+    try {
+      let primaryKeyIndex = 0
+      const setText = Object.keys(row)
+        .map((key, index) => {
+          const currentIndex = index + 1
+          if (key === 'id') {
```

Comment:
why is this needed ? you can have id in where and set clause
```
const keys = Object.keys(row)
const primaryKey = keys.indexOf('id') + 1
const setText = keys.map((key, index) => `key=${index+1}`)
```

Reply:
@andreievg I also wanted to avoid updating the primary key but the code you added simplified things a lot. Implementing.

Reply:
Resolved with 0c77bb6